.dotnet/src/Custom/Images/GeneratedImage.Serialization.cs

```diff
@@ -0,0 +1,118 @@
+using OpenAI.ClientShared.Internal;
+using System;
+using System.ClientModel.Primitives;
+using System.Text.Json;
+
+namespace OpenAI.Images;
+
+public partial class GeneratedImage : IJsonModel<GeneratedImage>
+{
+    GeneratedImage IJsonModel<GeneratedImage>.Create(ref Utf8JsonReader reader, ModelReaderWriterOptions options)
+    {
+        var format = options.Format == "W" ? ((IPersistableModel<GeneratedImage>)this).GetFormatFromOptions(options) : options.Format;
```

Comment:
these 5 lines of code are repeated a lot. Should we have one helper somewhere?

Reply:
I'll abstract it. Most of it will hopefully move back into generated code in the not-too-distant term, but we're still going to have custom implementations and there's no good reason to not de-dupe it.

Reply:
PR updated, the internal static `ModelSerializationHelpers` class abstracts the repeated patterns across `IJsonModel<T>` and `IPersistableModel<T>`.